### DISCUSSION TITLE
How does the auth middleware work?

### DISCUSSION THREAD

**Discussion by dev-mirzabicer** [hagopj13/node-express-boilerplate#222] [Q&A]
I've been inspecting the code and I'm stuck at the auth middleware. I understand the most of it, but I don't understand the part where passport "provides" the user (at the line `req.user = user` in the verification function's returned function). What is the type of the "user" parameter? Is it a string or is it a mongoose User document?

**Comment by alibabayev0** [comment]
You can find your answer with checking ./src/config/passport.js Ln 17, Col 5. There is we set const user = User.findById(......),
after getting user from our docs, we set it to req.user to pass our controllers as user.
